lib/knapsack_pro/extensions/rspec_extension.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+module KnapsackPro
+  module Extensions
+    # Knapsack extension on top of RSpec methods.
+    #
+    # You should use methods with the `knapsack__` prefix.
+    # The methods should hide public and private RSpec methods, as well as a complex chain of internal RSpec methods.
+    # You don't break the Law of Demeter when you use a `knapsack__` method. You can use complex or chained internal RSpec methods that are hidden inside of a `knapsack__` method.
+    #
+    # Often, the structure of the `knapsack__` method is inspired by an existing RSpec method. The `knapsack__` method should have a similar structure to the original RSpec method.
+    # It should be easy to see a code difference when a new RSpec version is released.
+    # Please use permalinks to RSpec source code in your comments when you create a new `knapsack__` method.
+    module RSpecExtension
+      def self.setup!
+        RSpec::Core::World.prepend(World)
+        RSpec::Core::Runner.prepend(Runner)
+        RSpec::Core::Configuration.prepend(Configuration)
+      end
+
+      module World
+        # Based on `announce_filters`
+        # https://github.com/rspec/rspec-core/blob/f8c8880dabd8f0544a6f91d8d4c857c1bd8df903/lib/rspec/core/world.rb#L171
+        def knapsack__setup
+          fail_if_config_and_cli_options_invalid
+        end
+      end
+
+      module Runner
+        def knapsack__mark_setup_as_done
+          @knapsack__setup_done = true
+        end
+
+        def knapsack__setup_done?
+          !!@knapsack__setup_done
+        end
+
+        # Based on:
+        # https://github.com/rspec/rspec-core/blob/f8c8880dabd8f0544a6f91d8d4c857c1bd8df903/lib/rspec/core/runner.rb#L98
+        def knapsack__setup(stream_error = $stderr, stream_out = $stdout)
+          # Abstract from RSpec::Core::Runner#setup, since we do not need to set any filters or files at this point,
+          # and we do not want to let world.announce_filters be called, since it will print
+          # out `No examples found.` message.
+          configure(stream_error, stream_out)
+
+          world.knapsack__setup
+
+          knapsack__mark_setup_as_done
+        end
+
+        def knapsack__wants_to_quit?
+          world.wants_to_quit
+        end
+
+        # `rspec_is_quitting` was added in RSpec 3.11.0
+        # Changelog:
+        #   https://github.com/rspec/rspec-core/blob/f8c8880dabd8f0544a6f91d8d4c857c1bd8df903/Changelog.md#3110--2022-02-09
+        # PR:
+        #   https://github.com/rspec/rspec-core/pull/2926
+        # We support RSpec < 3.11.0.
```

Comment:
Sorry for my mistake in the comment section. I wanted to express that we want to support both old RSpec < 3.11.0 and everything else (RSpec >= 3.11.0).

I added the comment so that you could find the original source of changes. 

A customer reported a bug that we need to support old RSpec without the `rspec_is_quitting` method. That's why this code exists.

Reply:
I thought you wanted to have the comment to explain why there's a `world.respond_to?(:rspec_is_quitting)`. My suggestion is to just keep `# RSpec >= 3.11.0` because there's no need for the rest: I can go to RSpec and pull up that info.

I'd be fine without any comments too. It's clear that a `respond_to` underlines different versions of RSpec.

Reply:
+1 to zero comments for this one. The `#respond_to?` make it obvious we don't take it as a given. No need for specifics in our own codebase I don't think.

Reply:
An additional improvement I was wondering about:
We could run integration tests against different RSpec versions on CI (tests would fail if someone removes `world.respond_to?(:rspec_is_quitting)`. Right now we don't do it for unit test step in CI pipeline.